### DISCUSSION TITLE
Regular API additions requested to pass a transaction back to be signed and then returned as a signed transaction for broadcasting

### DISCUSSION THREAD

**Discussion by johnheenan** [btcpayserver/btcpayserver#6262] [Ideas / Feature Requests]
Currently BTCPay facilitates cold wallets through a UI with BTCPay vault. I assume this uses an undocumented API with BTCPay.

Can we have two addiitons to the regular API that complement the regular API to create signed transactions to bypass working through the vault?

1) Create an unsigned transcation, with a reference id, and pass it back through the API through a GET request
2) Submit a signed transaction, with its reference id, for broadcast through a POST request

This would give a lot more flexibility to build additional solutions.

**Comment by pavlenex** [comment]
Hi, what's the use-case here?

**Comment by Aeroverra** [comment]
@pavlenex  The main use case is for deeper integration into custom software. For example, I’m setting up a 2-of-2 multisig flow on an e-commerce platform where BTCPay acts as an invisible backend gateway. The idea is that the merchant and customer never actually see the BTCPay interface.

It really comes down to two big factors:

Legal Compliance In the US specifically, holding user keys often gets you labeled a "Money Transmitter." That classification requires expensive licensing ($100k+) and carries harsh penalties. By using a 2-of-2 setup, or just ensuring we don't hold enough keys to move funds unilaterally, we stay non-custodial. This API change allows us to maintain that legal safety net while still automating the flow.

Security This setup means internal staff can’t embezzle funds since we don't hold the full quorum of keys. Also, if a user's private key is compromised externally, the attacker is stuck because they would still need to authenticate with our platform to get the transaction broadcasted.

To be honest, I think having an endpoint to handle unsigned/signed transactions should almost be the default. Most people using the API for automation are likely in an environment where that separation would be best.

**Comment by Aeroverra** [comment]
@pavlenex Oh also, in the security space for platform hot wallets

Even in a 2-of-2 setup where I hold both keys, splitting them up allows my application to call out to a separate signing server. This is significantly safer because that signing server can enforce specific failsafes like value thresholds or rate limits.

If the main application gets compromised and tries to sign too many transactions at once (suspicious/abnormal behavior), the signing server simply stops signing. This architectural separation prevents a total drain and is something that should essentially be the default for any automated environment dealing with more than a small personal website.


Although I do think the original requestor is misunderstanding something and if not maybe I am. I don't see any reason to store the unsigned transaction with a reference Id. The primary feature needed here would be broadcasting the PSBT and the get request would be more of a quality of life feature to easily construct a PSBT.

**Comment by Aeroverra** [comment]
PR Created https://github.com/btcpayserver/btcpayserver/pull/7068
